commit 7878a5a4fcc5002e805c054730c4c5639c9d071d
Author: Mohamed Abbas <mabbas@linux.intel.com>
Date:   Thu Nov 29 11:10:13 2007 +0800

    iwlwifi: enhance WPA authenication stability
    
    This patch enhanced WPA authenication stability by avoiding scan
    immediately followed by association. We don't do any scanning right
    after association in next several seconds. This will allow WPA
    authentication to take place without any interruption.
    
    Signed-off-by: Mohamed Abbas <mabbas@linux.intel.com>
    Signed-off-by: Zhu Yi <yi.zhu@intel.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/iwlwifi/iwl-3945.h b/drivers/net/wireless/iwlwifi/iwl-3945.h
index 75cef3ed9511..1f82061221a1 100644
--- a/drivers/net/wireless/iwlwifi/iwl-3945.h
+++ b/drivers/net/wireless/iwlwifi/iwl-3945.h
@@ -752,6 +752,7 @@ struct iwl3945_priv {
 
 	/* Scan related variables */
 	unsigned long last_scan_jiffies;
+	unsigned long next_scan_jiffies;
 	unsigned long scan_start;
 	unsigned long scan_pass_start;
 	unsigned long scan_start_tsf;